### PR TITLE
Add sdk-registry config

### PR DESCRIPTION
**Pull request by Chaoba**

### CODE REVIEW

File: gradle/gradle-sdk-registry.gradle [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+}
+
+registry {
+  sdkName = POM_ARTIFACT_ID
+  production = true
+  snapshot = false
+  override = false
+  dryRun = true
+  publish = true
+  publications = ["MapboxAndroidPluginSdk"]
```
Comment (pengdev) [inline gradle/gradle-sdk-registry.gradle:53]: should be `MapboxAndroidMapsSdk`

File: gradle/gradle-sdk-registry.gradle [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+apply plugin: 'maven-publish'
+apply plugin: 'com.mapbox.sdkRegistry'
+apply from: file('../gradle/artifact-settings.gradle')
+
+version = project.ext.versionName
+group = project.ext.mapboxArtifactGroupId
+
+publishing {
+  publications {
+    MapboxAndroidMapsSdk(MavenPublication) {
```
Comment (pengdev) [inline gradle/gradle-sdk-registry.gradle:11]: same as the publication name here.

File: gradle/publish.gradle [hunk 3]
```diff
@@ -1,123 +1,79 @@
...
+task androidJavadocs(type: Javadoc) {
+  source = android.sourceSets.main.java.sourceFiles
+  classpath = files(android.bootClasspath)
+  failOnError = false
+}
 
-  task androidJavadocsJar(type: Jar, dependsOn: androidJavadocs) {
-    classifier = 'dokka'
+task androidJavadocsJar(type: Jar, dependsOn: androidJavadocs) {
+    classifier = 'javadoc'
```
Comment (pengdev) [inline gradle/publish.gradle:62]: why change dokka to javadoc?

File: plugin-offline/gradle.properties [hunk 4]
```diff
@@ -1,5 +1,6 @@
-VERSION_NAME=0.8.0-SNAPSHOT
+VERSION_NAME=0.8.0
```
Comment (pengdev) [inline plugin-offline/gradle.properties:1]: After publishing manually, this version name can be changed to 0.9.0-SNAPSHOT

File: plugin-traffic/gradle.properties [hunk 5]
```diff
@@ -2,4 +2,5 @@ VERSION_NAME=0.11.0-SNAPSHOT
 POM_ARTIFACT_ID=mapbox-android-plugin-traffic-v9
 POM_NAME=Mapbox Android Traffic Plugin
 POM_DESCRIPTION=Mapbox Android Traffic Plugin
-POM_PACKAGING=aar
\ No newline at end of file
+POM_GROUP_ID=com.mapbox.mapboxsdk
```
Comment (pengdev) [inline plugin-traffic/gradle.properties:5]: it seems the group id is the same for all the modules, we could have it in the artifact-settings.gradle

File: circle.yml [hunk 6]
```diff
@@ -81,7 +81,7 @@ jobs:
 # ------------------------------------------------------------------------------
   release:
     docker:
-      - image: mbgl/android-ndk-r19:8e91a7ebab
+      - image: mbgl/android-ndk-r21:769c31f1d8
```
Comment (pengdev) [inline circle.yml:84]: probably should use ndk-r21e?

File: ktx-mapbox-maps/gradle.properties [hunk 7]
```diff
@@ -2,4 +2,4 @@ VERSION_NAME=0.1.0-SNAPSHOT
 POM_ARTIFACT_ID=mapbox-android-sdk-ktx-v7
 POM_NAME=Mapbox Maps SDK for Android KTX
 POM_DESCRIPTION=A set of Kotlin extensions for the Mapbox Maps SDK for Android.
-POM_PACKAGING=aar
\ No newline at end of file
+REGISTRY_SDK_NAME=mapbox-android-ktx
```
Comment (pengdev) [inline ktx-mapbox-maps/gradle.properties:5]: is this the correct name? looks different than the other names.

**Review by pengdev — APPROVED** [review]
LGTM with 2 questions, please create a ticket to track the tail work of enabling release automation through CI.

File: circle.yml (hunk 6 above)
Comment (Chaoba) [reply circle.yml:84]: It's align with maps sdk v9

File: ktx-mapbox-maps/gradle.properties (hunk 7 above)
Comment (Chaoba) [reply ktx-mapbox-maps/gradle.properties:5]: It seems we have never released this before. Please correct me if I'm wrong. @tobrun

**Pull request merged**